### PR TITLE
Return EndEvent information in ProcessStartResponse

### PR DESCRIPTION
**Pull request by 5ebastianMeier**
Blocked by
* https://github.com/process-engine/consumer_api_contracts/pull/27
* https://github.com/process-engine/process_engine_contracts/pull/38

## What did you change?

This PR implements the changes necessary to support the additional information mentioned in https://github.com/process-engine/consumer_api_contracts/issues/25.

If a process was started awaiting any kind of `EndEvent`, the `ProcessStartResponsePayload` now contains the `FlowNodeId` of the `EndEvent` reached, as well as its token payload.

## How can others test the changes?

See https://github.com/process-engine/process_engine_meta/pull/99.

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [ ] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.)
- [x] I've merged the `develop` branch into my branch before finishing this PR.
- [x] I've **not added any other changes** than the ones described above.
- [x] I've mentioned all **PRs, which relate to this one**

### CODE REVIEW

File: src/consumer_api.ts [hunk 1]
```diff
@@ -151,18 +152,32 @@ export class ConsumerApiService implements IConsumerApiService {
...
+                                                                                              processModel,
+                                                                                              correlationId,
+                                                                                              endEventKey,
+                                                                                              payload.inputValues);
     } else {
-      this.executeProcessService.startAndAwaitEndEvent(executionContext, processModel, correlationId, payload.inputValues);
+      endEventReachedMessage = await this.executeProcessService.startAndAwaitEndEvent(executionContext,
+                                                                                      processModel,
+                                                                                      correlationId,
+                                                                                      payload.inputValues);
```
Comment (cmg-dev) [inline src/consumer_api.ts]: würde ich noch nach `executeProcessService` umbrechen.

File: src/consumer_api.ts [hunk 2]
```diff
@@ -151,18 +152,32 @@ export class ConsumerApiService implements IConsumerApiService {
...
+
     if (startCallbackType === StartCallbackType.CallbackOnProcessInstanceCreated) {
       this.executeProcessService.start(executionContext, processModel, correlationId, payload.inputValues);
     } else if (startCallbackType === StartCallbackType.CallbackOnEndEventReached && endEventKey) {
-      this.executeProcessService.startAndAwaitSpecificEndEvent(executionContext, processModel, correlationId, endEventKey, payload.inputValues);
+      endEventReachedMessage = await this.executeProcessService.startAndAwaitSpecificEndEvent(executionContext,
+                                                                                              processModel,
+                                                                                              correlationId,
+                                                                                              endEventKey,
+                                                                                              payload.inputValues);
```
Comment (cmg-dev) [inline src/consumer_api.ts]: würde ich noch nach `executeProcessService` umbrechen.

File: src/consumer_api.ts [hunk 3]
```diff
@@ -151,18 +152,32 @@ export class ConsumerApiService implements IConsumerApiService {
     const correlationId: string = payload.correlationId || uuid.v4();
     const processModel: Model.Types.Process = await this.processModelPersistence.getProcessModelById(processModelId);
 
+    let endEventReachedMessage: EndEventReachedMessage;
+
     if (startCallbackType === StartCallbackType.CallbackOnProcessInstanceCreated) {
       this.executeProcessService.start(executionContext, processModel, correlationId, payload.inputValues);
     } else if (startCallbackType === StartCallbackType.CallbackOnEndEventReached && endEventKey) {
```
Comment (cmg-dev) [inline src/consumer_api.ts]: hier könnte man das `if` noch refactoren.

File: src/consumer_api.ts [hunk 4]
```diff
@@ -151,18 +152,32 @@ export class ConsumerApiService implements IConsumerApiService {
     const correlationId: string = payload.correlationId || uuid.v4();
     const processModel: Model.Types.Process = await this.processModelPersistence.getProcessModelById(processModelId);
 
+    let endEventReachedMessage: EndEventReachedMessage;
+
     if (startCallbackType === StartCallbackType.CallbackOnProcessInstanceCreated) {
```
Comment (cmg-dev) [inline src/consumer_api.ts]: hier könnte man das `if` noch refactoren.

**Comment by 5ebastianMeier** [comment]
@cmg-dev should look better now. I didn't break right after the `executeService`, because that's a bit awkward to read. But I separated the if-blocks to favor early exit and moved the decision part to a private method.

**Pull request merged**